### PR TITLE
[CORE-979] Link to updated docs

### PR DESCRIPTION
**Pull request by oredavids**
[CORE-979]

Link portions of content to updated sections of the docs website

[CORE-979]: https://gruntwork.atlassian.net/browse/CORE-979?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: _data/steps.yml [hunk 1]
```diff
@@ -20,8 +20,7 @@
 
 - title: Deploy an end-to-end architecture
   description: |
-    You can assemble all the code from the IaC Library yourself by following our
-    <a href="/guides">Production Deployment Guides</a>, or you can have the Gruntwork team assemble a
+    You can assemble all the code from the IaC Library yourself or you can have the Gruntwork team assemble a
```
Comment (brikis98) [inline _data/steps.yml:23]: Perhaps update the link to the docs site? 

"You can assemble all the code from the IaC Library yourself by following our docs or you can have..."

File: pages/index/_learn-more.html [hunk 2]
```diff
@@ -27,10 +27,10 @@ <h2>Get access to the Infrastructure as Code Library.</h2>
     <img src="/assets/img/infrastructure-cube-icons/infrastructure-design-your-stack-grunty.png" />
   </div>
   <div class="col-xs-12 col-sm-7 col-sm-pull-5 col-md-6 col-md-pull-6 col-description">
-    <h2>Deploy your entire tech stack.</h2>
+    <h2>Deploy your entire tech stack.</h2> <!-- PR Review Callout: Should this header change given its content vs the section above? -->
```
Comment (brikis98) [inline pages/index/_learn-more.html:30]: Not sure I understand the question?

File: pages/infrastructure-as-code-library/_library.html [hunk 3]
```diff
@@ -5,7 +5,7 @@ <h2>What's in the Library?</h2>
   infrastructure code for AWS, written in Terraform, Go, Bash,
   and Python. Check out
   <a
-    href="/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/"
```
Comment (brikis98) [inline pages/infrastructure-as-code-library/_library.html:8]: IIRC, this guide is linked to from a bunch of other places: e.g., module READMEs. So fixing it here alone won't be enough. We really need to ensure we have redirects in place!

File: pages/index/_learn-more.html (hunk 2 above)
Comment (oredavids) [reply pages/index/_learn-more.html:30]: After trying to re-state my question, I realize it is not really a concern. I will remove the comment.

**Review by zackproser — APPROVED** [review]
LGTM

**Pull request merged**